tendermint/src/abci/request.rs

```diff
@@ -0,0 +1,306 @@
+//! ABCI requests and request data.
+//!
+//! The [`Request`] enum records all possible ABCI requests. Requests that
+//! contain data are modeled as a separate struct, to avoid duplication of field
+//! definitions.
+
+use crate::prelude::*;
+
+// IMPORTANT NOTE ON DOCUMENTATION:
+//
+// The documentation for each request type is adapted from the ABCI Methods and
+// Types spec document. However, the same logical request may appear three
+// times, as a struct with the request data, as a Request variant, and as a
+// CategoryRequest variant.
+//
+// To avoid duplication, this documentation is stored in the doc/ folder in
+// individual .md files, which are pasted onto the relevant items using #[doc =
+// include_str!(...)].
+//
+// This is also why certain submodules have #[allow(unused)] imports to bring
+// items into scope for doc links, rather than changing the doc links -- it
+// allows the doc comments to be copied without editing.
+
+use core::convert::{TryFrom, TryInto};
+
+use super::MethodKind;
+
+// bring into scope for doc links
+#[allow(unused)]
+use super::types::Snapshot;
+
+mod apply_snapshot_chunk;
+mod begin_block;
+mod check_tx;
+mod deliver_tx;
+mod echo;
+mod end_block;
+mod info;
+mod init_chain;
+mod load_snapshot_chunk;
+mod offer_snapshot;
+mod query;
+
+pub use apply_snapshot_chunk::ApplySnapshotChunk;
+pub use begin_block::BeginBlock;
+pub use check_tx::{CheckTx, CheckTxKind};
+pub use deliver_tx::DeliverTx;
+pub use echo::Echo;
+pub use end_block::EndBlock;
+pub use info::Info;
+pub use init_chain::InitChain;
+pub use load_snapshot_chunk::LoadSnapshotChunk;
+pub use offer_snapshot::OfferSnapshot;
+pub use query::Query;
+
+/// All possible ABCI requests.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum Request {
+    #[doc = include_str!("doc/request-echo.md")]
```

Comment:
It looks like the CI failures are because the CI is hardcoding a version of nightly 6 months older than the stable version it uses. I think the fix would be to either remove the nightly-dependent CI check, or to make it always use the latest nightly.